presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -0,0 +1,253 @@
+    }
+    auto rowType = ROW(std::move(childNames), std::move(childTypes));
+    const size_t vectorSize = children.empty() ? 0 : children.front()->size();
+
+    return std::make_shared<RowVector>(
+        pool_.get(), rowType, BufferPtr(nullptr), vectorSize, children);
+  }
+
+  RowVectorPtr rowVector(
+      const std::shared_ptr<const RowType>& rowType,
+      vector_size_t size) {
+    return std::make_shared<RowVector>(
+        pool_.get(),
+        rowType,
+        BufferPtr(nullptr),
+        size,
+        std::vector<VectorPtr>{});
+  }
+
+  RowVectorPtr makeRowVector(
+      const std::vector<VectorPtr>& children,
+      std::function<bool(vector_size_t /*row*/)> isNullAt = nullptr) {
+    auto rowVec = rowVector(children);
+    if (isNullAt) {
+      setNulls(rowVec, isNullAt);
+    }
+    return rowVec;
+  }
+
+  void assertEqualVectors(const VectorPtr& expected, const VectorPtr& actual) {
+    ASSERT_EQ(expected->size(), actual->size());
+    ASSERT_TRUE(expected->type()->equivalent(*actual->type()))
+        << "Expected " << expected->type()->toString() << ", but got "
+        << actual->type()->toString();
+    for (auto i = 0; i < expected->size(); i++) {
+      ASSERT_TRUE(expected->equalValueAt(actual.get(), i, i))
+          << "at " << i << ": expected " << expected->toString(i)
+          << ", but got " << actual->toString(i);
+    }
+  }
+};
+
+TEST_F(UnsaferowShuffleTest, partitionAndSerializeOperator) {
+  exec::Operator::registerOperator(
+      std::make_unique<PartitionAndSerializeTranslator>());
+
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(1'000, [](auto row) { return row; }),
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row * 10; }),
+  });
+
+  auto plan =
+      exec::test::PlanBuilder()
+          .values({data}, true)
+          .addNode(addPartitionAndSerializeNode(4))
+          .planNode();
+
+  exec::test::CursorParameters params;
+  params.planNode = plan;
+  params.maxDrivers = 2;
+
+  auto [taskCursor, serializedResults] =
+      readCursor(params, [](auto /*task*/) {});
+  ASSERT_EQ(serializedResults.size(), 2);
+
+  for (auto& serializedResult : serializedResults) {
+    // Verify that serialized data can be deserialized successfully into the
+    // original data.
+    auto deserialized = deserialize(serializedResult, asRowType(data->type()));
+    assertEqualVectors(data, deserialized);
+  }
+}
+
+TEST_F(UnsafeRowShuffleTest, partitionAndSerializeStringTest) {
+  exec::Operator::registerOperator(
+      std::make_unique<PartitionAndSerializeTranslator>());
+
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(1'000, [](auto row) { return row; }),
+      makeFlatVector<int64_t>(1'000, [](auto row) { return row * 10; }),
+  });
+
+  auto plan =
+      exec::test::PlanBuilder()
+          .values({data}, true)
+          .addNode(addPartitionAndSerializeNode(4))
+          .planNode();
+
+  ASSERT_EQ(
+      plan->toString(true, false),
```

Comment:
Would you also add a test case with toString(false, false) and toString(true, true)?